worker-workflow/src/main/java/com/github/cafdataprocessing/workflow/WorkflowSettingsRetriever.java

```diff
@@ -0,0 +1,264 @@
+import com.github.cafdataprocessing.processing.service.client.model.EffectiveRepositoryConfigValue;
+import com.github.cafdataprocessing.processing.service.client.model.EffectiveTenantConfigValue;
+import com.github.cafdataprocessing.workflow.cache.WorkflowSettingsCacheKey;
+import com.github.cafdataprocessing.workflow.cache.WorkflowSettingsRepositoryCacheKey;
+import com.github.cafdataprocessing.workflow.cache.WorkflowSettingsTenantCacheKey;
+import static com.github.cafdataprocessing.workflow.transform.TransformerFunctions.LOG;
+import com.github.cafdataprocessing.workflow.transform.models.RepoConfigSource;
+import com.github.cafdataprocessing.workflow.transform.models.WorkflowSettings;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.gson.Gson;
+import com.hpe.caf.worker.document.exceptions.DocumentWorkerTransientException;
+import com.hpe.caf.worker.document.model.Document;
+import com.sun.jersey.api.client.ClientHandlerException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class WorkflowSettingsRetriever
+{
+    private final static Logger LOGGER = LoggerFactory.getLogger(WorkflowSettingsRetriever.class);
+    private final LoadingCache<WorkflowSettingsCacheKey, String> settingsCache;
+    private final ApiClient apiClient;
+    private final Gson gson;
+    private final RepositoryConfigurationApi repositoryConfigApi;
+    private final TenantConfigurationApi tenantConfigApi;
+
+    public WorkflowSettingsRetriever()
+    {
+        apiClient = new ApiClient();
+        settingsCache = CacheBuilder.newBuilder()
+            .maximumSize(100)
+            .expireAfterAccess(5, TimeUnit.MINUTES)
+            .recordStats()
+            .build(
+                new CacheLoader<WorkflowSettingsCacheKey, String>()
+            {
+                @Override
+                public String load(final WorkflowSettingsCacheKey key)
+                    throws ApiException, DocumentWorkerTransientException
+                {
+                    LOGGER.debug("Key not found in cache: " + key);
+                    return getSettingsFromServer(key);
+                }
+            });
+        gson = new Gson();
+        final String processingServiceBaseUrl = System.getenv("CAF_WORKFLOW_WORKER_PROCESSING_API_URL");
+        Objects.requireNonNull(processingServiceBaseUrl);
+        apiClient.setBasePath(processingServiceBaseUrl);
+        tenantConfigApi = new TenantConfigurationApi(apiClient);
+        repositoryConfigApi = new RepositoryConfigurationApi(apiClient);
+    }
+
+    public void addCustomWorkflowConfig(final WorkflowSettings requiredConfig, final Document document)
```

Comment:
Unused method

Reply:
This is the main entrypoint to the class